File: лабораторная-работа-1/smartcamera.cpp
#include "smartcamera.h"
#include <iostream>
using namespace std;

SmartCamera::SmartCamera(string name, float power, bool recording, int resolution)
    : SmartDevice(name, power), isRecording(recording), resolution(resolution) 
{}

SmartCamera::SmartCamera(string name, bool recording)
    : SmartDevice(name, 5), isRecording(recording), resolution(720) 
{}

SmartCamera::SmartCamera()
    : SmartDevice("Камера", 5), isRecording(false), resolution(720) 
{}

void SmartCamera::turnOn() {
    isPowered = true;
    cout << deviceName << " (камера) включена. ";
    if (isRecording) {
        cout << "Запись активна." << endl;
    } else {
        cout << "Запись неактивна." << endl;
    }
}

void SmartCamera::turnOff() {
    isPowered = false;
    cout << deviceName << " (камера) выключена." << endl;
}

void SmartCamera::displayInfo() {
    cout << "Устройство: " << deviceName 
         << " | Тип: Камера"
         << " | Статус: " << (isPowered ? "ВКЛ" : "ВЫКЛ")
         << " | Потребление: " << powerConsumption << " Вт"
         << " | Запись: " << (isRecording ? "ДА" : "НЕТ")
         << " | Разрешение: " << resolution << "p" << endl;
}

void SmartCamera::startRecording() {
    if (!isPowered) {
        cout << deviceName << ": Сначала включите камеру!" << endl;
        return;
    }
    isRecording = true;
    cout << deviceName << ": Запись начата в " << resolution << "p" << endl;
}

void SmartCamera::stopRecording() {
    isRecording = false;
    cout << deviceName << ": Запись остановлена." << endl;
}

File: лабораторная работа 2/SmartCamera.cs
using System;

namespace SmartHomeApp.Models
{
    public class SmartCamera : SmartDevice
    {
        private bool _isRecording;
        private int _resolution;

        public SmartCamera(string deviceName, float powerConsumption, bool isRecording, int resolution)
            : base(deviceName, powerConsumption)
        {
            _isRecording = isRecording;
            Resolution = resolution;
        }

        public SmartCamera(int id, string deviceName, float powerConsumption, bool isPowered, bool isRecording, int resolution)
            : base(id, deviceName, powerConsumption, isPowered)
        {
            _isRecording = isRecording;
            Resolution = resolution;
        }

        public bool IsRecording
        {
            get { return _isRecording; }
            set { _isRecording = value; }
        }

        public int Resolution
        {
            get { return _resolution; }
            set 
            { 
                if (value <= 480)
                    _resolution = 480;
                else if (value <= 720)
                    _resolution = 720;
                else if (value <= 1080)
                    _resolution = 1080;
                else
                    _resolution = 2160;
            }
        }

        public override void TurnOn()
        {
            IsPowered = true;
        }

        public override void TurnOff()
        {
            IsPowered = false;
            _isRecording = false;
        }

        public override string DisplayInfo()
        {
            string status = IsPowered ? "Включен" : "Выключен";
            string recordStatus = _isRecording ? "Идет запись" : "Запись не ведется";
            return $"Камера: {DeviceName}\nСтатус: {status}\nМощность: {PowerConsumption} Вт\nЗапись: {recordStatus}\nРазрешение: {_resolution}p";
        }

        public void StartRecording()
        {
            if (IsPowered)
            {
                _isRecording = true;
            }
        }

        public void StopRecording()
        {
            _isRecording = false;
        }
    }
}